### ISSUE TITLE
SQL `update` with JOOQ returns correctly updated field but doesn't write into DB

### ISSUE THREAD

**Issue by marcomicera** [fasten-project/fasten#291]
### TL;DR

1. The [`license-feeder` plugin](https://github.com/fasten-project/fasten/tree/7b932453f98e55f9fece99cadf25ae0f0a2a1b9d/analyzer/license-feeder) inserts outbound licenses into the `package_versions.metadata` field;
2. JOOQ returns the [correctly updated value of such field](https://github.com/fasten-project/fasten/blob/7b932453f98e55f9fece99cadf25ae0f0a2a1b9d/core/src/main/java/eu/fasten/core/data/metadatadb/MetadataDao.java#L244-L251) but it doesn't actually write into the DB.

## Describe the bug
This `update` query successfully inserts licenses into the DB int the `package_versions` table:
https://github.com/fasten-project/fasten/blob/7b932453f98e55f9fece99cadf25ae0f0a2a1b9d/core/src/main/java/eu/fasten/core/data/metadatadb/MetadataDao.java#L244-L251
`updatedMetadata ` contains a correctly updated value of the `pv.metadata` field: however, upon querying the DB later by my own, I don't see those licenses being appended to the existing `jsonb` value.

---

I tried to replicate this in a [test](https://github.com/fasten-project/fasten/blob/7b932453f98e55f9fece99cadf25ae0f0a2a1b9d/core/src/test/java/eu/fasten/core/data/metadatadb/MetadataDaoTest.java#L58-L161): not only it checks whether the returned metadata field has been successfully updated or not (which it is), but it also queries the database again, double-checking whether licenses have actually been inserted or not:
https://github.com/fasten-project/fasten/blob/7b932453f98e55f9fece99cadf25ae0f0a2a1b9d/core/src/test/java/eu/fasten/core/data/metadatadb/MetadataDaoTest.java#L140-L159
The weird thing is that this test passes (for four inputs: [1](https://github.com/fasten-project/fasten/runs/3123477876?check_suite_focus=true#step:4:3845), [2](https://github.com/fasten-project/fasten/runs/3123477876?check_suite_focus=true#step:4:3875), [3](https://github.com/fasten-project/fasten/runs/3123477876?check_suite_focus=true#step:4:3905), and [4](https://github.com/fasten-project/fasten/runs/3123477876?check_suite_focus=true#step:4:3935)).

## To Reproduce
1. Switch to the [`license-detector` branch](https://github.com/fasten-project/fasten-docker-deployment/tree/license-detector) of our docker-compose repo:
    ```shell script
    git clone --depth 1 --branch license-detector git@github.com:fasten-project/fasten-docker-deployment.git && cd fasten-docker-deployment
    ```
2. Start the `license-detection` profile:
    ```shell script
    docker-compose --profile license-detection up -d
    ```
3. Feed the pipeline with a Maven coordinate:
    ```shell script
    sed -n '1p' < test-resources/license-detector/valid-urls/mvn-projects.txt  | docker-compose exec -T kafka kafka-console-producer.sh --broker-list kafka:19092 --topic fasten.mvn.releases --property 'parse.key=true' --property 'key.separator=|'
    ```
4. Observe the [`license-feeder` plugin](https://github.com/fasten-project/fasten/tree/7b932453f98e55f9fece99cadf25ae0f0a2a1b9d/analyzer/license-feeder) and wait for it to insert outbound licenses at the package level:
    ```shell script
    docker-compose logs --follow fasten-license-feeder
    ```
5. Manually query the DB
    ```shell script
    PGPASSWORD=${FASTEN_METADATA_DB_PASSWORD} psql --host localhost --port 5432 --username fasten fasten_java --command='select metadata from package_versions;'
    ```

## Expected behavior
The `metadata` field should contain outbound licenses even when queried manually later in time.

## Additional context
It used to work before https://github.com/fasten-project/fasten/commit/70ca0e8835d12f902252c82c50b1f5f365eb81e9.

**Comment by marcomicera** [comment]
@MihhailSokolov I see you developed most of the [`MetadataDao` class](https://github.com/fasten-project/fasten/blob/7b932453f98e55f9fece99cadf25ae0f0a2a1b9d/core/src/main/java/eu/fasten/core/data/metadatadb/MetadataDao.java): did this ever happen to you?
This paragraph is enough:

> ### TL;DR
> 1. The [`license-feeder` plugin](https://github.com/fasten-project/fasten/tree/7b932453f98e55f9fece99cadf25ae0f0a2a1b9d/analyzer/license-feeder) inserts outbound licenses into the `package_versions.metadata` field;
> 2. JOOQ returns the [correctly updated value of such field](https://github.com/fasten-project/fasten/blob/7b932453f98e55f9fece99cadf25ae0f0a2a1b9d/core/src/main/java/eu/fasten/core/data/metadatadb/MetadataDao.java#L244-L251) but it doesn't actually write into the DB.

I'm using [`DSLContext.fetchValue(String sql)`](https://www.jooq.org/javadoc/3.14.x/org.jooq/org/jooq/DSLContext.html#fetchValue(java.lang.String)), I hope I'm using the wrong method and that you encountered this as well!

**Comment by MihhailSokolov** [comment]
@marcomicera I don't think I've come across this issue before. I would suggest that you try updating the metadata field in the same fashion it is done for callables - https://github.com/fasten-project/fasten/blob/c5180d2617193e5e4c0e528017f8b6dd8c91e9e6/core/src/main/java/eu/fasten/core/data/metadatadb/MetadataDao.java#L452

**Comment by marcomicera** [comment]
> @marcomicera I don't think I've come across this issue before. I would suggest that you try updating the metadata field in the same fashion it is done for callables -

Thanks for the immediate reply! I'll try this immediately.
I also didn't use plain SQL queries, but JOOQ was automatically converting my `jsonb` input as a `VARCHAR` when concatenating `||`, causing issues:
https://github.com/fasten-project/fasten/blob/b48bb8583cd7044a9f5869abf163a07c117eafde/core/src/main/java/eu/fasten/core/data/metadatadb/MetadataDao.java#L228-L243
But you don't seem to have this problem! I'll let you know.

**Comment by marcomicera** [comment]
The original [`insertPackageOutboundLicenses()` method](https://github.com/fasten-project/fasten/blob/7b932453f98e55f9fece99cadf25ae0f0a2a1b9d/core/src/main/java/eu/fasten/core/data/metadatadb/MetadataDao.java#L218-L256) writes successfully into the DB when calling it manually from Java:

```java
new MetadataDao(DSL.using(DriverManager.getConnection(url, username, password), SQLDialect.POSTGRES))
        .insertPackageOutboundLicenses(
                "com.esotericsoftware:reflectasm",
                "1.11.8",
                "{\"outbound\": [" +
                        "{" +
                        "\"name\": \"" + "apache-2.0" + "\"," +
                        "\"source\": \"" + DetectedLicenseSource.LOCAL_POM + "\"" +
                        "}"
                        + "]}"
        );
```

That's why tests are passing.
I must be doing something wrong with docker-compose, then...

**Comment by marcomicera** [comment]
> I must be doing something wrong with docker-compose, then...

Moving this to its dedicated repo...
